File: OpenWeatherCS/Services/DialogService.cs
using System.Windows;

namespace OpenWeatherCS.Services
{
    public class DialogService : IDialogService
    {
        public bool ShowConfirmationRequest(string message, string caption = "")
        {
            var result = MessageBox.Show(message, caption, MessageBoxButton.OKCancel);
            return result.HasFlag(MessageBoxResult.OK);
        }

        public void ShowNotification(string message, string caption = "")
        {
            MessageBox.Show(message, caption);
        }
    }
}

File: OpenWeather/Services/DialogService.vb
Public Class DialogService
    Implements IDialogService

    Public Sub ShowNotification(message As String, Optional caption As String = "") Implements IDialogService.ShowNotification
        MessageBox.Show(message, caption)
    End Sub

    Public Function ShowConfirmationRequest(message As String, Optional caption As String = "") As Boolean Implements IDialogService.ShowConfirmationRequest
        Dim result = MessageBox.Show(message, caption, MessageBoxButton.OKCancel)
        Return result.HasFlag(MessageBoxResult.OK)
    End Function
End Class
